### PR TITLE
[runtime] Make it an LLVM failure to try to compile unsupported wrapper types

### PR DESCRIPTION
**Pull request by alexanderkyte**

### CODE REVIEW

File: mono/mini/mini.h [hunk 1]
```diff
@@ -3191,4 +3191,10 @@ void mono_exception_native_unwind (void *ctx, MONO_SIG_HANDLER_INFO_TYPE *info);
  */
 void mono_interruption_checkpoint_from_trampoline (void);
 
+/*
+ * Get the name of a wrapper
+ */
+const char*
+get_wrapper_type_name (int type)
```
Comment (vargaz) [inline mono/mini/mini.h]: This should have a mono_ or mini_ prefix, preferably the latter.

File: mono/mini/mini-llvm.c [hunk 2]
```diff
@@ -3131,6 +3131,67 @@ emit_entry_bb (EmitContext *ctx, LLVMBuilderRef builder)
 }
 
 static void
+assert_only_accepted_wrappers (EmitContext *ctx, MonoMethod *method)
+{
+	gboolean FAIL = FALSE;
+
+	switch (method->wrapper_type) {
```
Comment (vargaz) [inline mono/mini/mini-llvm.c]: This should do a if (!ctx->llvm_only) return to skip the rest.

File: mono/mini/mini-llvm.c [hunk 3]
```diff
@@ -3131,6 +3131,67 @@ emit_entry_bb (EmitContext *ctx, LLVMBuilderRef builder)
 }
 
 static void
+assert_only_accepted_wrappers (EmitContext *ctx, MonoMethod *method)
+{
+	gboolean FAIL = FALSE;
+
```
Comment (vargaz) [inline mono/mini/mini-llvm.c]: This should be 'fail'.

File: mono/mini/mini-llvm.c [hunk 4]
```diff
@@ -3131,6 +3131,67 @@ emit_entry_bb (EmitContext *ctx, LLVMBuilderRef builder)
 }
 
 static void
+assert_only_accepted_wrappers (EmitContext *ctx, MonoMethod *method)
+{
```
Comment (vargaz) [inline mono/mini/mini-llvm.c]: This is a little misnamed, it doesn't actually assert in any way.

File: mono/mini/mini-llvm.c [hunk 5]
```diff
@@ -3154,6 +3215,9 @@ process_call (EmitContext *ctx, MonoBasicBlock *bb, LLVMBuilderRef *builder_ref,
 		return;
 	}
 
+	if (call->method != NULL)
+		assert_only_accepted_wrappers (ctx, call->method);
```
Comment (vargaz) [inline mono/mini/mini-llvm.c]: This should be if (call->method)

File: mono/mini/mini.h (hunk 1 above)
Comment (alexanderkyte) [inline mono/mini/mini.h]: Github isn't showing the update, but this has become 

mini_get_wrapper_type_name

**Comment by vargaz** [comment]
Why is this needed ? Why is it better to not compile these methods ?

**Comment by alexanderkyte** [comment]
A compile-time warning prevents the need to triage crashes that resemble ones caused by having these wrappers around. It's a debugging aid primarily. Although this will also decrease the footprint a bit.

**Pull request closed without merging**